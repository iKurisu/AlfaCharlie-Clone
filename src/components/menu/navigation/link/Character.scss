@import "sass/_variables.scss";

@media (max-width: $m) {
  .menu-nav-link:nth-child(3) .link-character:nth-child(1) {
    padding-left: 8px;
  }
}