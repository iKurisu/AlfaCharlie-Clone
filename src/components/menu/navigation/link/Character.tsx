import React, { useRef } from "react";
import useTransition from "hooks/useTransition";
import useDidUpdateEffect from "hooks/useDidUpdateEffect";

interface Props {
  char: string;
  order: number;
  isOpen: boolean;
}

const Character = ({ char, order, isOpen }: Props): JSX.Element => {
  const character = useRef(null);

  const fadeIn = useTransition(character, {
    from: { opacity: 0 },
    to: { opacity: 1 },
    config: {
      delay: 800 + order * 100,
      duration: 1000,
      timing: [0.28, 1, 0.5, 1]
    }
  });

  const reset = useTransition(character, {
    from: { opacity: 1 },
    to: { opacity: 0 },
    config: {
      delay: 1000,
      duration: 0
    }
  });

  useDidUpdateEffect((): void => {
    if (isOpen) {
      fadeIn();
    } else {
      reset();
    }
  }, [isOpen]);

  return (
    <span className="link-character" ref={character} style={{ opacity: 0 }}>
      {char}
    </span>
  );
};

export default Character;
