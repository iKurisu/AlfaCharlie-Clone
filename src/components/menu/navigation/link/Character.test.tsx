import React from "react";
import renderer from "react-test-renderer";
import Character from "./Character";

it("renders correctly", (): void => {
  const char = renderer.create(
    <Character char={"d"} order={1} isOpen={true} />
  );
  expect(char).toMatchSnapshot();
});
