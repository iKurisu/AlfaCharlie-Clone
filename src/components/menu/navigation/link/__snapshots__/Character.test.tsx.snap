// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="link-character"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  d
</span>
`;
